qds_sdk/connection.py

```diff
@@ -188,15 +188,15 @@ def _handle_error(response):
         elif code == 422:
             sys.stderr.write(response.text + "\n")
             raise ResourceInvalid(response)
-        elif code in (502, 503, 504):
+        elif code in (502, 504):
             sys.stderr.write(response.text + "\n")
             raise RetryWithDelay(response)
         elif code == 449:
             sys.stderr.write(response.text + "\n")
-            raise RetryWithDelay(response, "Data requested is unavailable. Retrying...")
-        elif code == 429:
+            raise RetryWithDelay(response, self.get_error_message(code))
+        elif code in (429, 503):
             sys.stderr.write(response.text + "\n")
-            raise ApiThrottledRetry(response, "Too many requests. Retrying...")
+            raise ApiThrottledRetry(response, self.get_error_message(code))
```

Comment:
If this is a static method, you don't need to use self here, just call Connection.method_name

Reply:
No need to change the def of handle error as self is not being used